libretroshare/src/gxs/rsgxsnetservice.cc

```diff
@@ -5268,35 +5279,87 @@ void RsGxsNetService::receiveTurtleSearchResults(TurtleRequestId req,const unsig
     mObserver->receiveNewGroups(new_grps);
 }
 
-bool RsGxsNetService::search(const std::string& substring,std::list<RsGxsGroupSummary>& group_infos)
+bool RsGxsNetService::search( const std::string& substring,
+                              std::list<RsGxsGroupSummary>& group_infos )
 {
+	group_infos.clear();
+
+#ifdef RS_DEEP_SEARCH
+	std::vector<DeepSearch::SearchResult> results;
+	DeepSearch::search(substring, results, 0);
+
+	for(auto dsr : results)
+	{
+		RsUrl rUrl(dsr.mUrl);
+		const auto& uQ(rUrl.query());
+		auto rit = uQ.find("id");
+		if(rit != rUrl.query().end())
+		{
+			RsGroupNetworkStats stats;
+			RsGxsGroupId grpId(rit->second);
+			if( !grpId.isNull() && getGroupNetworkStats(grpId, stats) )
+			{
+				RsGxsGroupSummary s;
+
+				s.mGroupId = grpId;
+
+				if((rit = uQ.find("name")) != uQ.end())
+					s.mGroupName = rit->second;
+				if((rit = uQ.find("signFlags")) != uQ.end())
+					s.mSignFlags = std::stoul(rit->second);
+				if((rit = uQ.find("publishDate")) != uQ.end())
+				{
+					std::tm tm; memset(&tm, 0, sizeof(tm));
+#if defined(GCC_VERSION) && GCC_VERSION > 50100
```

Comment:
It would probably be useful to have a function like RsUtil::rs_gettime() to avoid this #define thing everytime we use it.